core/src/main/java/de/evoila/cf/broker/service/impl/CatalogServiceImpl.java

```diff
@@ -144,4 +149,47 @@ public void filterActivePlans(Catalog catalog) {
 		});
 	}
 
+	public void controllAndManageMaintenanceInfo(Catalog catalog) {
```

Comment:
What does this method do?

Reply:
Okay now I got it.
Can we structure this method so its easier to read. 
We number of nested ifs and for is way to high.

Reply:
I think streams would help.

Reply:
now using streams, where possible